### PR TITLE
cypress: init at 3.2.0

### PR DESCRIPTION
**Pull request by acyuta108**
###### Motivation for this change
Missing package.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (aanderse) [inline maintainers/maintainer-list.nix]: Separate commit please.

File: pkgs/development/web/cypress/default.nix
Comment (ryantm) [inline pkgs/development/web/cypress/default.nix]:
```suggestion
  pname = "cypress";
```

File: pkgs/development/web/cypress/default.nix
Comment (ryantm) [inline pkgs/development/web/cypress/default.nix]: @erosennin @thorstenweber83 are you okay with being maintainers of this package?

**Review by ryantm — CHANGES_REQUESTED** [review]

File: pkgs/development/web/cypress/default.nix
Comment (thorstenweber83) [reply pkgs/development/web/cypress/default.nix]: Yeah, sure!

**Comment by macalinao** [comment]
Hey @acyuta108 , thanks for making this package. I'm getting the following error:

```
EACCES: permission denied, open '/home/igm/.config/Cypress/cy/production/browsers/chrome/interactive/CypressExtension/background.js'
```

It looks like it's `chmod`ded to 0700, and it belongs to my user. Any ideas? Thanks!

**Comment by acyuta108** [comment]
Yes, I am getting the same error. Unfortunately I believe is a Cypress issue. I discovered it last Friday, when I last tested right after my push and got the error. 
Try switching the browser to Electron and let me know if you still have the issue. 
Unfortunately if We don't figure this out before this package is to be merged, I'll need to downgrade to Cypress 3.1.5, which worked just fine.
I have [created a ticket](https://github.com/cypress-io/cypress/issues/3852) in Cypress to see if they can help.
@thorstenweber83 and @erosennin, you guys are far more experience than me, would you kindly take a look and see if something might have changed from 3.1.5 => 3.2.0 that we need to address when unpacking/unzipping and linking the contents of the deb package from Cypress?
Thanks

**Comment by macalinao** [comment]
Yes it works with Electron. I can also confirm the same error happens when
using Chromium.

File: pkgs/development/web/cypress/default.nix
Comment (acyuta108) [reply pkgs/development/web/cypress/default.nix]: @ryantm Sure, I'll make the change as requested. I haven't found any documentation on how to maintain multiple versions of a package. In this case, We might need to move forward with 3.1.5 until We resolve the latest issue in 3.2.0 as reported by @macalinao. Can you point me in the right direction? Thanks.

Comment (ryantm) [reply pkgs/development/web/cypress/default.nix]: The simplest thing is something like:

```
cypress = callPackage ./development/web/cypress/default.nix { }
cypress_3_1 = callPackage ./development/web/cypress/3.1.nix { }
```
Where `cypress` is the latest version and `cypress_3_1` is the earlier version needed by some specific package. Each would have their own derivation file to avoid breaking eachother.

**Comment by mmahut** [comment]
Any progress on this?

**Pull request closed without merging**
